### ISSUE TITLE
How much vram needed for controlnet training?

### ISSUE THREAD

**Issue by universewill** [XLabs-AI/x-flux#38]
How much vram needed for controlnet training?

**Comment by kadirnar** [comment]
You can train with A100.

**Comment by resindraburiza** [comment]
is 40GB A100 enough? or do you need the 80GB version?

**Comment by kadirnar** [comment]
> is 40GB A100 enough? or do you need the 80GB version?

You need 53GB Vram. So you should use A100 80GB. I am doing development to optimize it to work with 48GB vram. It is very difficult.  I think we should train it with Multi-GPU but it doesn't work. Training takes 3-4 days. If you train with default parameters it takes 2 weeks

**Comment by lkwq007** [comment]
> is 40GB A100 enough? or do you need the 80GB version?

With gradient checkpointing & batch_size=1, the vram usage is 39325MiB

**Comment by kadirnar** [comment]
> > is 40GB A100 enough? or do you need the 80GB version?
> 
> With gradient checkpointing & batch_size=1, the vram usage is 39325MiB

I set the batch size value to 1 but it didn't work with 48gb vram. Did you test it?

**Comment by lkwq007** [comment]
> > > is 40GB A100 enough? or do you need the 80GB version?
> > 
> > 
> > With gradient checkpointing & batch_size=1, the vram usage is 39325MiB
> 
> I set the batch size value to 1 but it didn't work with 48gb vram. Did you test it?

Tested with A40. **Gradient checkpointing** is required

![image](https://github.com/user-attachments/assets/eb568dc2-ea78-457e-9454-d1caa2880055)

**Comment by kadirnar** [comment]
> > > > is 40GB A100 enough? or do you need the 80GB version?
> > > 
> > > 
> > > With gradient checkpointing & batch_size=1, the vram usage is 39325MiB
> > 
> > 
> > I set the batch size value to 1 but it didn't work with 48gb vram. Did you test it?
> 
> Tested with A40. **Gradient checkpointing** is required
> 
> ![image](https://private-user-images.githubusercontent.com/1665437/359863931-eb568dc2-ea78-457e-9454-d1caa2880055.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQyNDA1NjMsIm5iZiI6MTcyNDI0MDI2MywicGF0aCI6Ii8xNjY1NDM3LzM1OTg2MzkzMS1lYjU2OGRjMi1lYTc4LTQ1N2UtOTQ1NC1kMWNhYTI4ODAwNTUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDgyMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA4MjFUMTEzNzQzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MDNlOWVmZjY4ZGVmMWE4ZDAzZjBjOTIzODIzYzA3ZjU5ZDdhNzM0YWIxYmVkOTk1NGMyYzAzNjY5YzQ4ZjM1MyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.P5l7CYbALtmZvKMMZINZffDMYtbI3SrW5hzC6N92nJw)

Can you share your config settings?

**Comment by kadirnar** [comment]
I tried it with 2xA40 and I get a cuda memory error.

<img width="1144" alt="image" src="https://github.com/user-attachments/assets/59ace17e-1fbe-4b03-adf2-202aaf97b7ba">

**Comment by ethxnp** [comment]
@lkwq007 can you please share your config?

**Comment by log26** [comment]
> > > > is 40GB A100 enough? or do you need the 80GB version?
> > > 
> > > 
> > > With gradient checkpointing & batch_size=1, the vram usage is 39325MiB
> > 
> > 
> > I set the batch size value to 1 but it didn't work with 48gb vram. Did you test it?
> 
> Tested with A40. **Gradient checkpointing** is required
> 
> ![image](https://private-user-images.githubusercontent.com/1665437/359863931-eb568dc2-ea78-457e-9454-d1caa2880055.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQ0MTM5OTYsIm5iZiI6MTcyNDQxMzY5NiwicGF0aCI6Ii8xNjY1NDM3LzM1OTg2MzkzMS1lYjU2OGRjMi1lYTc4LTQ1N2UtOTQ1NC1kMWNhYTI4ODAwNTUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDgyMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA4MjNUMTE0ODE2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NzYxZGU1NzY3MjBhMjFmMDUxZTgxNmIyM2MyYzFmZjEzOTllMTE5YjYwZTczM2FiNjQyNzk5ZDkwNmNiNWJkYiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.36TLpLU7Vy3rsglulqaGmOIl48Dj630SSf8Sc5a7eTY)

What is the img_size?

**Comment by kadirnar** [comment]
> > > > > is 40GB A100 enough? or do you need the 80GB version?
> > > > 
> > > > 
> > > > With gradient checkpointing & batch_size=1, the vram usage is 39325MiB
> > > 
> > > 
> > > I set the batch size value to 1 but it didn't work with 48gb vram. Did you test it?
> > 
> > 
> > Tested with A40. **Gradient checkpointing** is required
> > ![image](https://private-user-images.githubusercontent.com/1665437/359863931-eb568dc2-ea78-457e-9454-d1caa2880055.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQ0MTM5OTYsIm5iZiI6MTcyNDQxMzY5NiwicGF0aCI6Ii8xNjY1NDM3LzM1OTg2MzkzMS1lYjU2OGRjMi1lYTc4LTQ1N2UtOTQ1NC1kMWNhYTI4ODAwNTUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDgyMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA4MjNUMTE0ODE2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NzYxZGU1NzY3MjBhMjFmMDUxZTgxNmIyM2MyYzFmZjEzOTllMTE5YjYwZTczM2FiNjQyNzk5ZDkwNmNiNWJkYiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.36TLpLU7Vy3rsglulqaGmOIl48Dj630SSf8Sc5a7eTY)
> 
> What is the img_size?

1024

**Comment by iariav** [comment]
@kadirnar can you please share your config with img_size=1024?
